docs/how-to/debug/debug-with-gdb.rst

```diff
@@ -2,3 +2,224 @@
 
 Debug with gdb
 ==============
+
+The `GNU Debugger`_ (gdb) is used by developers to introspect the execution environment
+of an application, revealing both its code and data state at any point. Developers can
+use gdb to debug snaps locally and remotely.
+
+
+Local debugging with gdb
+------------------------
+
+Snaps are run within a :ref:`confined environment <reference-confinement>`, so running
+gdb directly against a snap executable would introspect both the snapd runtime
```

Comment:
> executable

We use this multiple times throughout this doc. Wouldn't "app" be better?

Reply:
There are some places where "binary" is slightly more precise, but using "app" everywhere is better for consistency. Updated.